From 5f619b2611e03ed2a4f3f21e8f9b10f345213685 Mon Sep 17 00:00:00 2001
From: Mladen Kovacevic <mladkov@gmail.com>
Date: Tue, 14 Jul 2015 15:56:44 -0400
Subject: [PATCH 095/140] KITE-1042: Add support for Oozie conf in CLI.

OOZIE_ACTION_CONF_XML added to kite-dataset along with 'config' option
that can be specified to be passed to the hadoop jar command for kite.
---
 .../kite-tools/src/main/sh/dataset.sh              |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/kite-tools-parent/kite-tools/src/main/sh/dataset.sh b/kite-tools-parent/kite-tools/src/main/sh/dataset.sh
index 88a4fe1..671789f 100644
--- a/kite-tools-parent/kite-tools/src/main/sh/dataset.sh
+++ b/kite-tools-parent/kite-tools/src/main/sh/dataset.sh
@@ -116,6 +116,17 @@ if [ ! -z "$KITE_USER_CLASSPATH" ]; then
 fi
 debug "Using HADOOP_CLASSPATH=${HADOOP_CLASSPATH}"
 
+OPT_OOZIE_ACTION_XML="-conf ${OOZIE_ACTION_CONF_XML}"
+if [ -z ${OOZIE_ACTION_CONF_XML} ]; then
+    # If the environment variable is not set, then neither do we set this
+    # value as a config variable
+    # The variable should only be set when this executable is called
+    # from within an Oozie shell action, in which case Oozie itself
+    # will set this variable.
+    OPT_OOZIE_ACTION_XML=""
+fi
+debug "OPT_OOZIE_ACTION_XML=${OPT_OOZIE_ACTION_XML}"
+
 export HADOOP_CLASSPATH
 export HADOOP_COMMON_HOME
 export HADOOP_MAPRED_HOME
@@ -124,7 +135,7 @@ export HIVE_CONF_DIR
 export HIVE_HOME
 
 if [ -x "$HADOOP_COMMON_HOME/bin/hadoop" ]; then
-  exec ${HADOOP_COMMON_HOME}/bin/hadoop jar "$0" $flags org.kitesdk.cli.Main --dollar-zero "$0" "$@"
+  exec ${HADOOP_COMMON_HOME}/bin/hadoop jar "$0" $flags org.kitesdk.cli.Main ${OPT_OOZIE_ACTION_XML} $config --dollar-zero "$0" "$@"
 else
   echo "ERROR: Cannot find Hadoop installation!"
   echo "You can fix this warning by setting HADOOP_HOME"
-- 
1.7.9.5

